features/step_definitions/test_steps.rb

```diff
@@ -227,4 +227,21 @@
 
 Then(/^I print all the text contents in the page using html tag$/) do
   puts browser.html
+end
+
+Then (/^I hover over the text "([^"]*)"$/) do |arg|
+  browser.span(:text => arg).when_present.hover
+end
+
+Then (/^I print the subcategories of class "([^"]*)"$/) do |classname|
+  browser.div(:class => classname).spans(:class => "nav-text").each do |span|
```

Comment:
Can you change `nav-text` to `nav-subtext` and check? You can remove the `browser.div(:class => classname)` code. Currently, it is not printing the right texts either.

Just put

```
browser.spans(:class => classname).each do |span|
puts span.text
```

where `classname` is `nav-subtext`

`nav-text` appears to be the links of the subcategories.
`nav-subtext` appears to be the texts present under the link.

The empty spaces, you are seeing in the result, is because, watir is not having the code to capture images that is existing in the popup.

You can try coding with `nav-text` and `nav-subtext`  wherever applicable with just `spans` and not `div`